src/composition/supergraph/version.rs

```diff
@@ -43,6 +46,72 @@ impl TryFrom<SupergraphVersion> for FederationVersion {
     }
 }
 
+lazy_static::lazy_static! {
+    static ref LATEST_PLUGIN_VERSIONS: Value = serde_json::from_str(include_str!("../../../latest_plugin_versions.json")).expect("could not read latest_plugin_versions.json from the root of the repo, which is needed to supply latest versions to `rover supergraph compsoe`.");
+
+}
+
+/// FederationVersion is the apollo_federation_types's view of the version of federation (ie, the
+/// spec and its implementation by Apollo) in use. This can be an exact version or point to the
+/// latest of a major version (eg, latest of version 1, latest of version 2). The
+/// SupergraphVersion, however, is the version of the supergraph binary. These are synonymous, but
+/// different; FederationVersion can be inexact by pointing to the latest of some major version
+/// while SupergraphVersion must be exact because we must use an exact version of the binary
+///
+/// Development note: when we have latest-*, we not only get an exact version, we get an exact
+/// version specified in our latest_plugins_versions.json. This version might be different than the
+/// actual latest version if we haven't updated that file
+impl TryFrom<FederationVersion> for SupergraphVersion {
+    type Error = SupergraphVersionError;
+    fn try_from(federation_version: FederationVersion) -> Result<Self, Self::Error> {
+        let supergraph = LATEST_PLUGIN_VERSIONS["supergraph"]
+            .as_object()
+            .expect("JSON malformed: top-level `supergraph` should be an object");
+
+        let supergraph_versions = supergraph
+            .get("versions")
+            .expect("JSON malformed: `supergraph.versions` did not exist");
```

Comment:
There are a lot of calls to `expect()` in this fn - do we want to return errors?